src/api/users/test/e2e/users.test.js

```diff
@@ -206,192 +85,101 @@ describe('PUT REQUESTS', () => {
       isFlagged: true,
       feeds: ['https://dev.to/feed/galileogalilei'],
       github: {
-        username: 'galileogalilei',
+        username: 'GalileoGalilei',
         avatarUrl:
           'https://avatars.githubusercontent.com/u/7242003?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
       },
     };
-
-    const response = await request(app)
-      .put(`/${user.id}`)
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      .send(updated);
-
-    expect(response.statusCode).toBe(200);
-  });
-
-  test('Rejected - Ensure that the id param has to match the hashed email of the body', async () => {
-    const { user } = await createUser();
-    // Get the original id (based on current email)
-    const { id } = user;
-    // Change the email
-    user.email = 'modified@email.com';
-
-    const response = await request(app)
-      // Use the original id
-      .put(`/${id}`)
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      // But the newly updated data, with modified email
-      .send(user);
-
-    expect(response.statusCode).toBe(400);
-  });
-
-  test('Rejected - Update a nonexistent user', async () => {
-    const user = new User(defaultUsers.carl());
-
-    const response = await request(app)
-      .put(`/${user.id}`)
-      .set('Content-Type', 'application/json')
-      .set('Authorization', `bearer ${createServiceToken()}`)
-      .send(user);
-
-    expect(response.statusCode).toBe(404);
   });
-});
-
-describe('POST REQUESTS', () => {
-  beforeEach(clearData);
 
-  test('Accepted - Create a user', async () => {
-    const { response } = await createUser();
-    expect(response.statusCode).toBe(201);
-  });
+  it('should not post a user with missing first name info', async () => {
+    delete galileo.firstName;
+    const response = await postUser(galileo);
+    const parsedResponse = await response.json();
 
-  test('Rejected - Create two of the same user', async () => {
-    const { response: response1 } = await createUser();
-    expect(response1.statusCode).toBe(201);
+    expect(parsedResponse.statusCode).toBe(400);
+    expect(parsedResponse.validation.body.message).toBe('"firstName" is required');
```

Comment:
For the future, this is an anti-pattern.  You want to depend on error codes, not error messages, because the error message might change (or get translated to another language, etc) but the error code will not.

Reply:
Sounds good, thanks for the tip! For reference, this message we're testing is a message returned by celebrate rather than a specific response message.